### PR TITLE
feat: allow specifcation of full url for endpoint

### PR DESCRIPTION
**Pull request by rsmitty**
This PR moves to using the full URL for endpoint instead of trying to
hardcode 6443 in various places like we were doing.

Signed-off-by: Spencer Smith <robertspencersmith@gmail.com>

### CODE REVIEW

File: pkg/config/types/v1alpha1/generate/generate.go [hunk 1]
```diff
@@ -123,7 +124,15 @@ func (i *Input) GetControlPlaneEndpoint() string {
 // GetAPIServerSANs returns the formatted list of Subject Alt Name addresses for the API Server
 func (i *Input) GetAPIServerSANs() []string {
 	list := []string{"127.0.0.1", "::1"}
-	list = append(list, i.ControlPlaneEndpoint)
+
+	endpointURL, err := url.Parse(i.ControlPlaneEndpoint)
```
Comment (andrewrynhard) [inline pkg/config/types/v1alpha1/generate/generate.go]: Not necessarily for this PR, but I think it makes sense to make `ControlPlaneEndpoint` a `*net.URL`.

File: internal/app/machined/internal/phase/services/post_boot.go [hunk 2]
```diff
@@ -35,9 +34,11 @@ func (task *LabelNodeAsMaster) standard(r runtime.Runtime) (err error) {
...
 	}
 
-	endpoint := net.ParseIP(r.Config().Cluster().Endpoint())
-
-	h, err := kubernetes.NewTemporaryClientFromPKI(r.Config().Cluster().CA().Crt, r.Config().Cluster().CA().Key, endpoint.String(), "6443")
+	h, err := kubernetes.NewTemporaryClientFromPKI(
+		r.Config().Cluster().CA().Crt,
+		r.Config().Cluster().CA().Key,
+		r.Config().Cluster().Endpoint().Hostname(),
+		r.Config().Cluster().Endpoint().Port())
```
Comment (andrewrynhard) [inline internal/app/machined/internal/phase/services/post_boot.go]: nit:
```suggestion
		r.Config().Cluster().Endpoint().Port(),
	)
```

Comment (rsmitty) [reply internal/app/machined/internal/phase/services/post_boot.go]: Fixed in amended commit

File: pkg/config/types/v1alpha1/generate/generate.go (hunk 1 above)
Comment (rsmitty) [reply pkg/config/types/v1alpha1/generate/generate.go]: totally agree. I'll drop a backlog ticket for this

**Pull request merged**